From 1ca1fb11588e6254d7c89ce277328a5186d7567a Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 11 Jul 2014 13:38:37 +0100
Subject: [PATCH 047/118] mllib: Create a common utility function is_directory.

This is a wrapper around Sys.is_directory which doesn't throw
exceptions.

(cherry picked from commit 893bfe2e36f5a766e822b14fe5e7e4fc9b2b3f75)
---
 mllib/common_utils.ml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mllib/common_utils.ml b/mllib/common_utils.ml
index c3a9382..77f2d31 100644
--- a/mllib/common_utils.ml
+++ b/mllib/common_utils.ml
@@ -412,3 +412,10 @@ let is_block_device file =
 let is_char_device file =
   try (Unix.stat file).Unix.st_kind = Unix.S_CHR
   with Unix.Unix_error _ -> false
+
+(* Annoyingly Sys.is_directory throws an exception on failure
+ * (RHBZ#1022431).
+ *)
+let is_directory path =
+  try Sys.is_directory path
+  with Sys_error _ -> false
-- 
2.7.4

